enhancements/management-workload-partitioning.md

```diff
@@ -0,0 +1,549 @@
+   section? Should we add a `managementWorkloadPartitioning` section,
+   or something similar, and say for now that it only applies when the
+   single-node deployment approach is used?
+2. What should the new field in the Infrastructure CRD be?
+
+### Test Plan
+
+We will add a CI job to ensure that all release payload workloads and
+their namespaces are labeled with `io.openshift.management: true`.
+
+We will add a CI job to ensure that single-node deployments configured
+with management workload partitioning pass the compliance tests.
+
+### Graduation Criteria
+
+**Note:** *Section not required until targeted at a release.*
+
+Define graduation milestones.
+
+These may be defined in terms of API maturity, or as something else. Initial proposal
+should keep this high-level with a focus on what signals will be looked at to
+determine graduation.
+
+Consider the following in developing the graduation criteria for this
+enhancement:
+
+- Maturity levels
+  - [`alpha`, `beta`, `stable` in upstream Kubernetes][maturity-levels]
+  - `Dev Preview`, `Tech Preview`, `GA` in OpenShift
+- [Deprecation policy][deprecation-policy]
+
+Clearly define what graduation means by either linking to the [API doc definition](https://kubernetes.io/docs/concepts/overview/kubernetes-api/#api-versioning),
+or by redefining what graduation means.
+
+In general, we try to use the same stages (alpha, beta, GA), regardless how the functionality is accessed.
+
+[maturity-levels]: https://git.k8s.io/community/contributors/devel/sig-architecture/api_changes.md#alpha-beta-and-stable-versions
+[deprecation-policy]: https://kubernetes.io/docs/reference/using-api/deprecation-policy/
+
+**Examples**: These are generalized examples to consider, in addition
+to the aforementioned [maturity levels][maturity-levels].
+
+#### Dev Preview -> Tech Preview
+
+- Ability to utilize the enhancement end to end
+- End user documentation, relative API stability
+- Sufficient test coverage
+- Gather feedback from users rather than just developers
+- Enumerate service level indicators (SLIs), expose SLIs as metrics
+- Write symptoms-based alerts for the component(s)
+
+#### Tech Preview -> GA
+
+- More testing (upgrade, downgrade, scale)
+- Sufficient time for feedback
+- Available by default
+- Backhaul SLI telemetry
+- Document SLOs for the component
+- Conduct load testing
+
+**For non-optional features moving to GA, the graduation criteria must include
+end to end tests.**
+
+#### Removing a deprecated feature
+
+- Announce deprecation and support policy of the existing feature
+- Deprecate the feature
+
+### Upgrade / Downgrade Strategy
+
+This new behavior will be added in 4.8 for single-node clusters only.
+
+Enabling the feature after installation is not supported, so we do not
+need to address what happens if an older cluster upgrades and then the
+feature is turned on.
+
+### Version Skew Strategy
+
+N/A
+
+## Implementation History
+
```

Comment:
https://github.com/openshift/kubernetes/pull/627 has the kubelet changes

https://github.com/cri-o/cri-o/pull/4680 has the first set of CRI-O changes